AlttpRandomizer/Rom/RomLocationsCasual.cs

```diff
@@ -103,11 +103,12 @@ public void ResetLocations()
                     CanAccess =
                         have =>
                         CanEnterSwampPalace(have) 
-                        && have.Contains(InventoryItemType.Hammer) 
-                        && (have.Contains(InventoryItemType.Hookshot) 
-                            || (!LocationHasItem("[dungeon-D2-B2] Swamp Palace - flooded room [left chest]",InventoryItemType.BigKey) 
-                                && !LocationHasItem("[dungeon-D2-B2] Swamp Palace - flooded room [right chest]",InventoryItemType.BigKey)   
-                                && !LocationHasItem("[dungeon-D2-B2] Swamp Palace - hidden waterfall door room",InventoryItemType.BigKey))),
+                        && have.Contains(InventoryItemType.Hammer)
+                        && ((LocationHasItem("[dungeon-D2-B1] Swamp Palace - big key room", InventoryItemType.BigKey)
+                                || LocationHasItem("[dungeon-D2-B1] Swamp Palace - map room", InventoryItemType.BigKey)
+                                || LocationHasItem("[dungeon-D2-B1] Swamp Palace - push 4 blocks room", InventoryItemType.BigKey)
+                                || LocationHasItem("[dungeon-D2-B1] Swamp Palace - south of hookshot room", InventoryItemType.BigKey))
+                            || (have.Contains(InventoryItemType.Hookshot) && !LocationHasItem("[dungeon-D2-B1] Swamp Palace - big chest",InventoryItemType.Hookshot)))

```

Comment:
Please follow the convention of having each case on a separate line. 

Looks like you have an unnecessary set of parens around those four rooms where you are checking for the big key (since they are followed by another ||.)